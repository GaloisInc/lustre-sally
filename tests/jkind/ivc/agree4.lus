-- IVC for __GUARANTEE0: _TOP__COMP~0.__COMP__PROP__2

node _TOP__COMP(
  in1 : int;
  in2 : int;
  out1 : int;
  out2 : int
) returns (
  __ASSERT : bool
);
var
  __ASSUME__HIST : bool;
  __ASSUME__CONJ : bool;
  __COMP__PROP__0 : bool;
  __COMP__PROP__1 : bool;
  __COMP__PROP__2 : bool;
let
  __ASSUME__CONJ = true;

  __ASSUME__HIST = (__ASSUME__CONJ -> ((pre __ASSUME__HIST) and __ASSUME__CONJ));

  __COMP__PROP__0 = (out1 > (-1));

  __COMP__PROP__1 = (out1 < 1);

  __COMP__PROP__2 = (out2 > 10);

  __ASSERT = (__ASSUME__HIST => (__COMP__PROP__2 and (__COMP__PROP__1 and (__COMP__PROP__0 and true))));

  --%IVC __COMP__PROP__0,__COMP__PROP__1,__COMP__PROP__2;

tel;

node main(
  COMP__in1 : int;
  COMP__in2 : int;
  COMP___CLOCK_ : bool;
  input1 : int;
  input2 : int;
  COMP__out1 : int;
  COMP__out2 : int;
  internal_var1 : int;
  internal_var2 : int;
  output1 : int;
  output2 : int;
  output3 : int;
  output4 : int
) returns (

);
var
  __GUARANTEE0 : bool;
let
  --%MAIN;
  __GUARANTEE0 = (output2 > (internal_var1 + internal_var2));

  assert condact(true, _TOP__COMP(COMP__in1, COMP__in2, COMP__out1, COMP__out2), true);

  assert (input1 = COMP__in1);

  assert (input2 = COMP__in2);

  assert (COMP__out1 = output1);

  assert (COMP__out2 = output2);

  assert (internal_var1 = 5);

  assert (internal_var2 = 5);

  assert true;

  --%PROPERTY __GUARANTEE0;

tel;
