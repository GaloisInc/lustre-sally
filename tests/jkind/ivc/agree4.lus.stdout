Sally Model
===========

(define-state-type
   S
   (
     (Eq Bool)
     (in1 Int)
     (in2 Int)
     (input1 Int)
     (input2 Int)
     (out1 Int)
     (out2 Int)
     (output1 Int)
     (output2 Int)
     (output3 Int)
     (output4 Int)
     (|COMP___CLOCK_| Bool)
     (|COMP__in1| Int)
     (|COMP__in2| Int)
     (|COMP__out1| Int)
     (|COMP__out2| Int)
     (|Eq:1| Bool)
     (|Eq:2| Bool)
     (|Eq:3| Bool)
     (|Eq:4| Bool)
     (|Eq:5| Bool)
     (|_TOP__COMP| Bool)
     (|__ASSUME__CONJ| Bool)
     (|__ASSUME__HIST:init| Bool)
     (|__ASSUME__HIST| Bool)
     (|__COMP__PROP__0| Bool)
     (|__COMP__PROP__1| Bool)
     (|__COMP__PROP__2| Bool)
     (|__GUARANTEE0| Bool)
     (|gal-initializing| Bool)
     (|internal_var1| Int)
     (|internal_var2| Int)
     (|pre___ASSUME__HIST| Bool))
   (
     (input1 Int)
     (input2 Int)
     (output1 Int)
     (output2 Int)
     (output3 Int)
     (output4 Int)
     (|COMP___CLOCK_| Bool)
     (|COMP__in1| Int)
     (|COMP__in2| Int)
     (|COMP__out1| Int)
     (|COMP__out2| Int)
     (|internal_var1| Int)
     (|internal_var2| Int)))
(define-transition-system
   TS
   S
   (and
      (= |gal-initializing| true)
      (= |__ASSUME__HIST:init| true))
   (and
      (= next.|gal-initializing| false)
      (= next.|COMP__in1| input.|COMP__in1|)
      (= next.|COMP__in2| input.|COMP__in2|)
      (= next.|COMP___CLOCK_| input.|COMP___CLOCK_|)
      (= next.input1 input.input1)
      (= next.input2 input.input2)
      (= next.|COMP__out1| input.|COMP__out1|)
      (= next.|COMP__out2| input.|COMP__out2|)
      (= next.|internal_var1| input.|internal_var1|)
      (= next.|internal_var2| input.|internal_var2|)
      (= next.output1 input.output1)
      (= next.output2 input.output2)
      (= next.output3 input.output3)
      (= next.output4 input.output4)
      next.Eq
      next.|Eq:1|
      next.|Eq:2|
      next.|Eq:3|
      next.|Eq:4|
      next.|Eq:5|
      next.|_TOP__COMP|
      (=
         next.|__GUARANTEE0|
         (<
            (+ next.|internal_var1| next.|internal_var2|)
            next.output2))
      (= next.|__ASSUME__CONJ| true)
      (= next.|pre___ASSUME__HIST| state.|__ASSUME__HIST|)
      (ite
         state.|__ASSUME__HIST:init|
         (= next.|__ASSUME__HIST| next.|__ASSUME__CONJ|)
         (=
            next.|__ASSUME__HIST|
            (and next.|pre___ASSUME__HIST| next.|__ASSUME__CONJ|)))
      (= next.|__ASSUME__HIST:init| false)
      (= next.out1 next.|COMP__out1|)
      (=
         next.|__COMP__PROP__0|
         (<
            (- 1)
            next.out1))
      (=
         next.|__COMP__PROP__1|
         (< next.out1 1))
      (= next.out2 next.|COMP__out2|)
      (=
         next.|__COMP__PROP__2|
         (< 10 next.out2))
      (=
         next.|_TOP__COMP|
         (=>
            next.|__ASSUME__HIST|
            (and next.|__COMP__PROP__2| next.|__COMP__PROP__1| next.|__COMP__PROP__0| true)))
      (= next.in1 next.|COMP__in1|)
      (= next.in2 next.|COMP__in2|)
      (=
         next.|Eq:5|
         (= next.input1 next.|COMP__in1|))
      (=
         next.|Eq:4|
         (= next.input2 next.|COMP__in2|))
      (=
         next.|Eq:3|
         (= next.|COMP__out1| next.output1))
      (=
         next.|Eq:2|
         (= next.|COMP__out2| next.output2))
      (=
         next.|Eq:1|
         (= next.|internal_var1| 5))
      (=
         next.Eq
         (= next.|internal_var2| 5))))

(query
   TS
   (or |gal-initializing| |__GUARANTEE0|))

[Lustre] Validating properties:
[Lustre] Property __GUARANTEE0...
  [Sally][Unknown]
  [Sally][Unknown]
[Lustre] Summary:
  [Valid] 0
  [Unknown] 1
  [Invalid] 0
[Lustre] Model status: [Unknown]
